src/ape/utils.py

```diff
@@ -513,45 +557,106 @@ def get_all_files_in_directory(path: Path) -> List[Path]:
     Returns:
         List[pathlib.Path]: A list of files in the given directory.
     """
+
     if path.is_dir():
         return list(path.rglob("*.*"))
 
     return [path]
 
 
-class AbstractDataClassMeta(DataClassMeta, ABCMeta):
+class injected_before_use(property):
+    """
+    Injected properties are injected class variables that must be set before use
+    NOTE: do not appear in a Pydantic model's set of properties
+    """
+
```

Comment:
Explore (in a new PR) refactoring this class to accept a `type_str` argument, and automatically doing the `cast` by some sort of integration with mypy (maybe using generics?)